servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,548 @@
+        requireNonNull(contextPath, "contextPath");
+        requireNonNull(level, "level");
+        this.contextPath = contextPath;
+        this.level = level;
+        if (contextPath.isEmpty()) {
+            servletContextName = "";
+        } else {
+            servletContextName = contextPath.substring(1).replace("/", "_").trim();
+        }
+    }
+
+    boolean isInitialized() {
+        return initialized;
+    }
+
+    /**
+     * Set server started.
+     */
+    void init() {
+        initialized = true;
+        initParamMap = ImmutableMap.copyOf(initParamMap);
+        servletRegistrations = ImmutableMap.copyOf(servletRegistrations);
+        sessionTrackingModeSet = ImmutableSet.copyOf(sessionTrackingModeSet);
+        mimeMappings = ImmutableMap.copyOf(mimeMappings);
+    }
+
+    void mimeMapping(String extension, String mimeType) {
+        requireNonNull(extension, "extension");
+        requireNonNull(mimeType, "mimeType");
+        mimeMappings.put(extension, mimeType);
+    }
+
+    void mimeMappings(Map<String, String> mappings) {
+        mimeMappings.putAll(requireNonNull(mappings, "mappings"));
+    }
+
+    @Override
+    public int getSessionTimeout() {
+        return sessionTimeoutMinutes;
+    }
+
+    @Override
+    public void setSessionTimeout(int sessionTimeout) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getContextPath() {
+        return contextPath;
+    }
+
+    @Override
+    public DefaultServletContext getContext(String uripath) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public int getMajorVersion() {
+        return 4;
+    }
+
+    @Override
+    public int getMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    public int getEffectiveMajorVersion() {
+        return 4;
+    }
+
+    @Override
+    public int getEffectiveMinorVersion() {
+        return 0;
+    }
+
+    @Override
+    @Nullable
+    public String getMimeType(@Nullable String file) {
+        if (file == null) {
+            return null;
+        }
+        final int period = file.lastIndexOf('.');
+        if (period < 0) {
+            return null;
+        }
+        final String extension = file.substring(period + 1);
+        if (extension.isEmpty()) {
+            return null;
+        }
+        return mimeMappings.get(extension);
+    }
+
+    @Override
+    public Set<String> getResourcePaths(String path) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public URL getResource(String path) throws MalformedURLException {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public InputStream getResourceAsStream(String path) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public String getRealPath(String path) {
+        throw new UnsupportedOperationException();
+    }
```

Comment:
IIRC, many many servlet implementations rely on these operations to load some resources such as templates and properties files, no?